apps/services/auth-public-api/src/app/modules/delegations/delegations.controller.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  DelegationsService,
+  IDelegation,
+  DelegationDTO,
+  Delegation,
+} from '@island.is/auth-api-lib'
+import {
+  IdsUserGuard,
+  Scopes,
+  ScopesGuard,
+  User,
+  CurrentUser,
+} from '@island.is/auth-nest-tools'
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Post,
+  Put,
+  UseGuards,
+} from '@nestjs/common'
+import { ApiCreatedResponse, ApiOkResponse, ApiTags } from '@nestjs/swagger'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@ApiTags('delegations')
+@Controller('delegations')
```

Comment:
It would be mighty nice if these endpoints created/updated/returned scopes as well, as these will generally be used together, and there might be a lot of scopes.

If you'd rather keep these separate, I'd at least like a scopeBulkUpdate endpoint that overrides the list of scopes for a delegation.

Reply:
Fyi, the design has a screen that lists "all" delegations, grouped by toNationalId, sorted by person's name, and listing which scopes each has.

This will be tricky to paginate without having a copy of user's names. But we should try not to do N+1 requests to show what each delegated user has access to.